tests/Tests.FeatureManagement/FeatureManagement.cs

```diff
@@ -306,6 +349,141 @@ public async Task Targeting()
+                {
+                    options.IgnoreMissingFeatureFilters = true;
+                });
+
+            services
+                .AddSingleton(config)
+                .AddFeatureManagement()
+                .AddFeatureVariantAssigner<ContextualTargetingFeatureVariantAssigner>();
+
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IFeatureVariantManager variantManager = serviceProvider.GetRequiredService<IFeatureVariantManager>();
+
+            //
+            // Targeted
+            Assert.Equal("def", await variantManager.GetVariantAsync<string, ITargetingContext>(
+                Features.ContextualVariantTargetingFeature,
+                new TargetingContext
+                {
+                    UserId = "Jeff"
+                },
+                CancellationToken.None));
+
+            //
+            // Not targeted
+            Assert.Equal("abc", await variantManager.GetVariantAsync<string, ITargetingContext>(
+                Features.ContextualVariantTargetingFeature,
+                new TargetingContext
+                {
+                    UserId = "Patty"
+                },
+                CancellationToken.None));
+        }
+
+        [Fact]
+        public async Task AccumulatesAudience()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+            var services = new ServiceCollection();
+
+            services
+                .Configure<FeatureManagementOptions>(options =>
+                {
+                    options.IgnoreMissingFeatureFilters = true;
+                });
+
+            services
+                .AddSingleton(config)
+                .AddFeatureManagement()
+                .AddFeatureVariantAssigner<ContextualTargetingFeatureVariantAssigner>();
+
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IFeatureVariantManager variantManager = serviceProvider.GetRequiredService<IFeatureVariantManager>();
+
+            var occurences = new Dictionary<string, int>();
+
+            //
+            // Test default rollout percentage accumulation
+            for (int i = 0; i < 3000; i++)
+            {
+                string result = await variantManager.GetVariantAsync<string, ITargetingContext>(
+                    "AccumulatedTargetingFeature",
+                    new TargetingContext
+                    {
+                        UserId = RandomHelper.GetRandomString(32)
+                    },
+                    CancellationToken.None);
+
+                if (!occurences.ContainsKey(result))
+                {
+                    occurences.Add(result, 1);
+                }
+                else
+                {
+                    occurences[result]++;
+                }
+            }
+
+            int total = occurences.Values.Aggregate((cur, t) => cur + t);
```

Comment:
Should the total be 3000 (the number of loops)?

Reply:
Yes.